gsad/src/gsad_http.c

```diff
@@ -804,22 +806,22 @@ attach_remove_sid (http_response_t *response, const gchar *sid)
 {
   if (sid)
     {
-      if (strcmp (sid, "0"))
+      if (str_equal (sid, "0"))
```

Comment:
I think there's another negation missing here.

Reply:
If have changed the sequence of the clauses. Should be fine here.